### PR TITLE
K8SPG-71 PGO operator containers

### PR DESCRIPTION
**Pull request by cap1984**
Since PG operator consists of 4 containers we need to mention all
of them

### CODE REVIEW

File: api/version.proto [hunk 1]
```diff
@@ -136,6 +136,10 @@ message VersionMatrix {
   map<string, Version> pgbouncer = 13;
   map<string, Version> pxc_operator = 14;
   map<string, Version> psmdb_operator = 15;
+  map<string, Version> operator_apiserver = 16;
+  map<string, Version> operator_event = 17;
+  map<string, Version> operator_rmdata = 18;
+  map<string, Version> operator_scheduler = 19;
```
Comment (hors) [inline api/version.proto:142]: operator_deployer ?

Comment (nonemax) [inline api/version.proto:142]: what about `deployer`?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

File: api/version.proto (hunk 1 above)
Comment (hors) [inline api/version.proto:142]: Maybe it is better to use something like that:
```suggestion
  map<string, Version> pg_operator_apiserver = 16;
  map<string, Version> pg_operator_event = 17;
  map<string, Version> pg_operator_rmdata = 18;
  map<string, Version> pg_operator_scheduler = 19;
  map<string, Version> pg_operator = 20;
  map<string, Version> pg_operator_deployer = 21;
```

Comment (cap1984) [reply api/version.proto:142]: I don't think it's needed here. @hors what do you think?

Comment (cap1984) [reply api/version.proto:142]: We can do that

Comment (cap1984) [reply api/version.proto]: done

Comment (cap1984) [reply api/version.proto]: done

Comment (cap1984) [reply api/version.proto]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

File: api-tests/apply_route_test.go [hunk 2]
```diff
@@ -115,11 +115,11 @@ func TestApplyPgShouldReturnSameMajorVersion(t *testing.T) {
 	pgParams := &version_service.VersionServiceApplyParams{
 		Apply:           "latest",
 		OperatorVersion: "1.0.0",
-		Product:         "postgresql-operator",
+		Product:         "pg-operator",
 	}
 	pgParams.WithTimeout(2 * time.Second)
 
-	for _, v := range []string{"11.0", "12.0", "13.0"} {
+	for _, v := range []string{"12.8", "13.4"} {
```
Comment (hors) [inline api-tests/apply_route_test.go:122]: we need to add 14.1, 13.5 and 12.9 (for next release)

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

File: sources/operator.1.0.0.pg-operator.json [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            "image_hash": "43108a08158006340a97378a79bedb56019743f9bbe01fbcdc7d058a36076c68",
+            "status": "recommended",
+            "critical": false
+          }
+        },
+        "pgbackrest": {
+          "12.8": {
+            "image_path": "percona/percona-postgresql-operator:1.0.0-ppg12-pgbackrest",
+            "image_hash": "1d8699e1820bc656a6fcae1bb3baac3912a6c917b27a5b57a3b539a85c0f5273",
+            "status": "available",
```
Comment (hors) [inline sources/operator.1.0.0.pg-operator.json:33]: we need to support 12-recommended, 13-recommended and 14-recommended options
please check it

File: api-tests/apply_route_test.go (hunk 2 above)
Comment (cap1984) [reply api-tests/apply_route_test.go:122]: Let's add it on exact release time

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

File: server/filters.go [hunk 4]
```diff
@@ -225,6 +225,51 @@ func defaultFilter(versions map[string]*pbVersion.Version, apply string) error {
...
+	if err != nil {
+		return "", status.Errorf(codes.Internal, "failed to sort versions: %v", err)
+	}
+
+	desired := sorted[0].String()
+	for _, s := range sorted {
+		versStr := s.String()
+		if strings.HasSuffix(versStr, ".0") {
+			versStr = strings.TrimSuffix(versStr, ".0")
+
```
Comment (egegunes) [inline server/filters.go:248]: nitpick but we should remove this empty line and put an empty line to L249

**Review by egegunes — DISMISSED** [review]
LGTM, just a small cosmetic comment

File: sources/operator.1.0.0.pg-operator.json [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+          "12.8": {
+            "image_path": "percona/percona-postgresql-operator:1.0.0-ppg12-pgbouncer",
+            "image_hash": "d52e4a335e9ed45ea55a9c96195d379c38bd28038cdc73df4b2645b513481c09",
+            "status": "recommended",
+            "critical": false
+          }
+        },
+        "operator": {
+          "1.0.0": {
+            "image_path": "percona/percona-postgresql-operator:1.0.0-postgres-operator",
```
Comment (delgod) [inline sources/operator.1.0.0.pg-operator.json:87]: maybe it has the sense to be consistent. up to @hors 
```suggestion
            "image_path": "percona/percona-postgresql-operator:1.0.0-pgo",
```

File: sources/operator.1.0.0.pg-operator.dep.json [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            },
+            "12.8"
+          ]
+        },
+        {
+          "<": [
+            {
+              "var": "productVersion"
+            },
+            "13.4"
```
Comment (delgod) [inline sources/operator.1.0.0.pg-operator.dep.json:26]:
```suggestion
            "13.0"
```

File: sources/operator.1.0.0.pg-operator.dep.json [hunk 7]
```diff
@@ -0,0 +1,122 @@
+{
+  "pgbackrest": {
+    "13.4": {
+      ">=": [
+        {
+          "var": "productVersion"
+        },
+        "13.4"
+      ]
```
Comment (delgod) [inline sources/operator.1.0.0.pg-operator.dep.json:9]: and < 14.0

File: server/filters.go [hunk 8]
```diff
@@ -225,6 +225,51 @@ func defaultFilter(versions map[string]*pbVersion.Version, apply string) error {
...
+	sorted, err := sortedVersionsDesc(keys)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, "failed to sort versions: %v", err)
+	}
+
+	desired := sorted[0].String()
+	for _, s := range sorted {
+		versStr := s.String()
+		if strings.HasSuffix(versStr, ".0") {
+			versStr = strings.TrimSuffix(versStr, ".0")
```
Comment (delgod) [inline server/filters.go:247]: operator version can be any, it does not always ends with `.0`.
can we avoid cutting `.0`?

**Review by delgod — CHANGES_REQUESTED** [review]

File: sources/operator.1.0.0.pg-operator.json (hunk 5 above)
Comment (hors) [reply sources/operator.1.0.0.pg-operator.json:87]: @delgod I think we can use new c for this image but only  for new version `1.1.0` :)    These files for `1.0.0` which was released more then one month ago.

File: server/filters.go (hunk 8 above)
Comment (nonemax) [reply server/filters.go:247]: it is about postgresql version, And we cut '.0' only  after we get 'version' var from 'semvar' package witch  always add '.0', so we have '13.4.0' and  so on

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| api-tests | passed |

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Review by nonemax — APPROVED** [review]

**Pull request merged**